### ISSUE TITLE
plugin metrics breaks application start up in Firefox

### ISSUE THREAD

**Issue by akosyakov** [eclipse-theia/theia#6582]
![Screen Shot 2019-11-19 at 14 42 50](https://user-images.githubusercontent.com/3082655/69151461-f7841c00-0ada-11ea-8b2b-3ae1dd4e7fd6.png)

cc @tsmaeder  @JPinkney

It seems FireFox cannot handle parsing some regex patterns. Maybe https://github.com/eclipse-theia/theia/blob/master/packages/plugin-metrics/src/browser/plugin-metrics-creator.ts#L29

**Comment by vince-fugnitto** [comment]
Lookbehind regexp are not supported in browsers besides `Chrome`.
The regexp will need to be updated.

**Comment by vince-fugnitto** [comment]
All non-chrome browsers likely do not start now (verified with Firefox and Safari).

**Comment by tsmaeder** [comment]
The regex in question is: `(?<=Request)(.*?)(?=failed)`. I don't think it contains anything that could not be rewritten without the negative lookahead.

**Comment by vince-fugnitto** [comment]
> The regex in question is: `(?<=Request)(.*?)(?=failed)`. I don't think it contains anything that could not be rewritten without the negative lookahead.

@tsmaeder would you like to provide a fix for it? I'm really not the best at regexp.
Also without tests, I'm not sure my implementation would be working correctly.

**Comment by tsmaeder** [comment]
I can have a look tomorrow, not time today.

**Comment by AndrienkoAleksandr** [comment]
@akosyakov Sorry for delay, I am starting testing this pr.

**Comment by tsmaeder** [comment]
@AndrienkoAleksandr wrong issue maybe? Which PR?

**Comment by AndrienkoAleksandr** [comment]
> @AndrienkoAleksandr wrong issue maybe? Which PR?

Sorry, wrong issue.

**Comment by tsmaeder** [comment]
I'm trying to test a fix, but I never get any failures if I test like described on https://github.com/eclipse-theia/theia/pull/6303

**Comment by tsmaeder** [comment]
@akosyakov @vince-fugnitto any ideas what I'm doing wrong?

**Comment by tsmaeder** [comment]
Hmh...all of a sudden, it works. Ignore my pleas!

**Comment by vince-fugnitto** [comment]
@tsmaeder I did see this when testing the original PR:

```
 language_server_time_count{id="redhat.vscode-yaml" method="textDocument/hover" result="fail"} 0 # HELP language_server_time_sum Sum of time in milliseconds that language server requests take # TYPE language_server_time_sum gauge language_server_time_sum{id="redhat.vscode-yaml" method="textDocument/hover" result="success"} 33.35499999229796 language_server_time_sum{id="redhat.vscode-yaml" method="textDocument/hover" result="failure"} 0
```

I'm not sure this enough or what to look for.

**Comment by tsmaeder** [comment]
Simply using /Request(.*)failed/ will not work, because of eager matching. I think I'll just cut off the prefix/postfix.

**Comment by vince-fugnitto** [comment]
Closed by #6591
